### DISCUSSION TITLE
How to get StatusCode (quality) of incoming data

### DISCUSSION THREAD

**Discussion by arunkumar790** [FreeOpcUa/python-opcua#1485] [Q&A] [answered]
Hi,
I have created a client and able to receive data by subscribing to data change. I wish to receive quality of the incoming data as it is. Based on information posted in the discussions here, it seems the possibility is by using StatusCode and severity field. I tried to implement it using def status_change_notification(self, status) but not achieved. Could you please help me with some more details

**Comment by AndreasHeine** [comment, marked as answer]
if you investigate
https://github.com/FreeOpcUa/python-opcua/blob/96a8ea672d09ab8bc78388a403998b0baa14fa95/opcua/common/subscription.py#L18

-> def datachange_notification(self, node, val, data):

data contain's the whole notification which has the DataValue (Value + StatusCode + Timestamps) you are looking for!

**Reply by arunkumar790** [reply]
Thanks a lot Andreas, i could able to get the quality status using data.monitored_item.Value.StatusCode
Really appreciate your support to this forum
